Sprint_7

nikkotin342 opened the pull request:
Добавлены методы и тесты финального тестирования
Добавлены общие файлы

Commit:
first_commit
Добавлены методы и тесты финального тестирования
Добавлены общие файлы

Commit:
Теперь точно добавлено все, что описано в первом коммите

nikkotin342 commented:
Прошу прощения! Вчера почему то автоматом галочки не проставились у файлов, и я только сейчас увидел, что запушил пустой коммит. Во втором коммите все подтянул, смотрите его пожалуйста

klushcheva commented on tests/couriers/test_create_courier.py:
```diff
@@ -0,0 +1,23 @@
+class TestCreateCourier:
+
+    def test_create_courier(self,courier_data):
+        response=CourierMethod().create_courier(*courier_data)
```
Нужно поправить: после создания курьера в любом тесте удаляем данные. Используй для этого фикстуру и метод удаления

klushcheva commented on tests/couriers/test_create_courier.py:
```diff
@@ -0,0 +1,23 @@
+
+    def test_create_courier_login_copy(self):
+        response=CourierMethod().create_courier('qwerty1','12345','qwerty1')
+        assert response[1] == {'code': 409, "message": 'Этот логин уже используется. Попробуйте другой.'}
```
Можно лучше: не хардкодить проверку, содержание message выносить в переменные (возможные тексты ошибок)
И проверять отдельно код ошибки

Commit:
Сообщения с ошибками убраны в файл data.py
test_create_courier.py - теперь удаляет созданного курьера с помощью новой фикстуры create_courier_and_delete

klushcheva commented on tests/couriers/conftest.py:
```diff
@@ -1,7 +1,23 @@
+    response=CourierMethod().create_courier(*courier_data)
+    response_lp = CourierMethod().login_courier(*response[2])
+    response_del = CourierMethod().delete_courier(response_lp[1]['id'])
+    return response[0], response[1]
```
Так тесты не будут проходить, так как ты удаляешь курьера перед тем, как начнется тест. Делаем так:

```
yield response[0], response[1]
CourierMethod().delete_courier(response_lp[1]['id'])
```

nikkotin342 replied:
Спасибо большое за помощь!

Commit:
Внесены изменения в фикстуру согласно ревью
